File: src/main/java/siosio/repository/ListComponentRef.java
package siosio.repository;

import com.intellij.util.xml.Attribute;
import com.intellij.util.xml.Convert;
import com.intellij.util.xml.DomElement;
import com.intellij.util.xml.GenericAttributeValue;
import org.jetbrains.annotations.NotNull;
import siosio.repository.converter.RepositoryListRefConverter;

/**
 * listタグ内に置かれるcomponent-refタグ
 */
public interface ListComponentRef extends DomElement {
    
    @NotNull
    @Attribute("name")
    @Convert(RepositoryListRefConverter.class)
    GenericAttributeValue<Component> getComponentClass();

}

File: src/main/java/siosio/repository/ListComponentRef.kt
package siosio.repository

import com.intellij.util.xml.*
import siosio.repository.converter.*

/**
 * listタグ内に置かれるcomponent-refタグ
 */
interface ListComponentRef : DomElement {

    @get:Attribute("name")
    @get:Convert(RepositoryListRefConverter::class)
    val componentClass: GenericAttributeValue<Component>

}
